torch/_dynamo/eval_frame.py

```diff
@@ -141,7 +142,20 @@ def __init__(
     ):
         super().__init__()
         assert callable(callback) or callback is False or callback is None
-        self.callback: DynamoCallback = callback
+
+        if sys.version_info >= (3, 11):
+            # This MUST be the inner most callback that is responsible for converting
+            # from the raw ptr back to the _PyInterpreterFrame object.
```

Comment:
It's weird we don't do this from C

Reply:
oh because _PyInterpreterFrame is complicated, sigh

Reply:
Yes it is. And we don't have a python binding for it.
I could manually write a python binding for that object in C but that didn't sounds that much cleaner tbh.

Reply:
I turned around and did it in C :D